sci-rs/src/signal/filter/savgol_filter.rs

```diff
@@ -7,6 +7,12 @@ use nalgebra as na;
 use nalgebra::RealField;
 use num_traits::Float;
 
+//for testing
+#[cfg(feature = "std")]
+use std::fs::OpenOptions;
+#[cfg(feature = "std")]
+use std::io::Write;
+
 #[cfg(feature = "alloc")]
```

Comment:
Imports related to testing should only be included in the test mods. That would be moving imports into the 
```rust
#[cfg(test)]
mod tests {
  use super::*; //  just for import crate stuff, not test deps

  ...

}
```

Separately, I don't think we want tests that edit the file system, so these imports will be unnecessary.